### ISSUE TITLE
Minecraft refuses to launch without Xwayland

### ISSUE THREAD

**Issue by figuld** [minecraft-linux/mcpelauncher-manifest#942]
**Describe the bug**
Minecraft always runs under Xwayland. If I disable it, it refuses to launch.

**To Reproduce**
1. Launch Minecraft.
2. `swaymsg -t get_tree`
3. `#11: con "Minecraft" (xwayland, pid: 13210, instance: "mcpelauncher-client", class: "mcpelauncher-client", X11 window: 0x800002)`
4. Uncomment `xwayland disable` in the sway config.
5. See crash.

**Expected behavior**
Minecraft should launch through wayland.

**Screenshots**
Game Log
```
15:31:55 Info  [Launcher] Version: client 4733750 / manifest 556dbe8
15:31:55 Info  [Launcher] CPU: GenuineIntel Intel(R) Celeron(R) N5100 @ 1.10GHz
15:31:55 Info  [Launcher] CPU supports SSSE3: YES
15:31:55 Trace [Launcher] Loading hybris libraries
15:31:55 Trace [Launcher] linker loaded
EGLUT: failed to initialize native display

Process exited with unexpected exit code: 1
```

**Desktop (please complete the following information):**
 - OS: Arch Linux
 - Version: `0.11.0.qt6.r13.g556dbe8-1`
 - Installation Type: chaotic-aur

**Additional context**
```
mcpelauncher-client 4733750 / manifest 556dbe8
CPU: GenuineIntel Intel(R) Celeron(R) N5100 @ 1.10GHz
SSSE3 support: YES
GL Vendor: Intel
GL Renderer: Mesa Intel(R) UHD Graphics (JSL)
GL Version: OpenGL ES 3.2 Mesa 23.3.4-arch1.2
MSA daemon path:
```

The launcher itself runs on wayland.
`#14: con "Linux Minecraft Launcher" (xdg_shell, pid: 13946, app_id: "io.mrarm.mcpelauncher-ui-qt")`

`sway version 1.8.1`
```
Name            : glfw-wayland
Version         : 3.3.9-1
Provides        : glfw=3.3.9
Depends On      : wayland  libxkbcommon  libgl
```
```
Name            : libglvnd
Version         : 1.7.0-1
Provides        : libgl  libegl  libgles  libEGL.so=1-64  libGL.so=1-64  libGLESv2.so=2-64  libGLX.so=0-64
                  libGLdispatch.so=0-64  libOpenGL.so=0-64
Depends On      : libxext  mesa  opengl-driver
```
```
Name            : freeglut
Version         : 3.4.0-1
Provides        : glut
Depends On      : libxi  libxrandr  libgl
Optional Deps   : None
Required By     : jasper  lib32-freeglut  libcaca  mesa-demos
Optional For    : libtiff
```

**Comment by figuld** [comment]
[Turns out, this is the freeglut issue.](https://github.com/benl23x5/gloss/issues/54#issuecomment-1880201305)

Is glut library required? (Can't find it in the dependency chains.) Maybe it's possible to use something else as another backend?

**Comment by ChristopherHX** [comment]
> Is glut library required?
 
Yes for a lot of prebuilt packages it is compiled in without opt out.

[eglut](https://github.com/minecraft-linux/eglut) has been required for touch support, because GLFW don't implement touch.


However I plan to do eglut ===> SDL3 (https://github.com/minecraft-linux/mcpelauncher-manifest/pull/935#issue-2092116930).

flatpak uses GLFW and runs in wayland only mode, but doesn't support touch.

**Comment by ChristopherHX** [comment]
> Uncomment xwayland disable in the sway config.

Why would you do that?

**Comment by ChristopherHX** [comment]
I'm not doing fast migration to SDL3, because of potential bugs in the backend.

**Comment by FreeFull** [comment]
Maybe this issue has been fixed now? Using the mpcelauncher flatpak, it runs and works fine for me without Xwayland

**Comment by ChristopherHX** [comment]
mcpelauncher flatpak is a pure wayland app via glfw even before this issue has been even opened

**Comment by ChristopherHX** [comment]
The xenial, bionic and buster AppImages will require XWayland for ever.

I will close this issue once bookworm AppImages use sdl3 by default that can take a while.
